### DISCUSSION TITLE
reactive form to filter the form choices below

### DISCUSSION THREAD

**Discussion by srmippili** [espanso/espanso#1927] [Feature requests and Ideas]
It would be great to make the form choices (the list is a big list) filterable based on another input in the form, similar to what we have for filtering all triggers using some keys, we don't have to remember any triggers.

Example, I try to open some files from a directory, initially I would show all files, and based on what the user keys in a keyword, the form should be able to show new filtered set of files. Right now this is missing and it would make this a great feature. 

Thanks a lot to providing some replacement of autohotkey for mac users. I am so relieved. 
Keep up the great work. Thanks

**Comment by srmippili** [comment]
@federico-terzi  - please see if it's possible to make the form like autocomplete . Looking fwd for your comment on this feature. It's very useful to create very generic search files plugin from espanso

**Comment by srmippili** [comment]
@federico-terzi please see this feature boss. It's very useful to make some generic triggers

**Comment by smeech** [comment]
Federico is no longer actively involved in the project but we have a small, but enthusiastic, team involved in developing various aspects and trying to fix existing limitations.

Your suggestion is a good one, and we'll keep this open but I'll close #1970 as a duplicate.

However, it's unlikely that we'll find time to implement this in the near future. If you have an urgent need I am happy to try and help script a way around it if you'd like to provide more specific details. Alternatively you could pose the question in the "ask-help" section on [Discord](https://discord.gg/4QARseMS6k).

**Reply by srmippili** [reply]
Thanks for the reply. Nothing urgent as such, but, is it possible to guide the script way for this? 
Since autocomplete is not available on forms, I am trying to take 2 or 3 static inputs, to filter the in context data. 
Any workaround via script way, do let me know. Thanks

**Reply by smeech** [reply]
I would envisage a refinement of the two-stage process along the lines of that documented in https://espanso.org/docs/matches/variables/#injecting-variables-inside-variables.
